pkg/reconciler/resource.go

```diff
@@ -122,20 +134,53 @@ type GenericResourceReconciler struct {
 
 type ResourceReconcilerOption func(*ReconcilerOpts)
 
+type RecreateResourceCondition func(kind schema.GroupVersionKind, status metav1.Status) bool
+
 // Recommended to use NewReconcilerWith + ResourceReconcilerOptions
 type ReconcilerOpts struct {
-	Log                                              logr.Logger
-	Scheme                                           *runtime.Scheme
-	EnableRecreateWorkloadOnImmutableFieldChange     bool
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+	// Enable recreating workloads and services when the API server rejects an update
+	EnableRecreateWorkloadOnImmutableFieldChange bool
+	// Custom log message to help when a workload or service need to be recreated
```

Comment:
I guess simple tense should be used in case of `or`, but I might be mistaken :)
```suggestion
	// Custom log message to help when a workload or service needs to be recreated
```